libraries/AP_Scripting/generator/src/main.c

```diff
@@ -970,53 +957,84 @@ void emit_userdata_allocators(void) {
     fprintf(source, "    luaL_getmetatable(L, \"%s\");\n", node->name);
     fprintf(source, "    lua_setmetatable(L, -2);\n");
     fprintf(source, "    return 1;\n");
-    fprintf(source, "}\n\n");
+    fprintf(source, "}\n");
+    if (node->dependency != NULL) {
+      fprintf(source, "#endif\n");
```

Comment:
```suggestion
      fprintf(source, "#endif // %s\n", node->dependency);
```
It's helpful sometimes (particularly when debugging) to have the comment indicate what we just finished checking

Reply:
Actually emitting this header/footer for dependencies is so common maybe we should have a helper for `start_dependency()` and `end_dependency()`? It's readable as is, I'm just noting that there seems to be a lot of repeated bits at the moment.

Reply:
moved to a `start_dependency()` and `end_dependency()` helpers